neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/BoltNeo4jDialect.java

```diff
@@ -589,6 +592,31 @@ public void forEachTuple(ModelConsumer consumer, TupleTypeContext tupleTypeConte
 		consumer.consume( tupleSupplier );
 	}
 
+	@Override
+	public ClosableIterator<Tuple> callStoredProcedure(
+			String storedProcedureName, ProcedureQueryParameters queryParameters, TupleContext tupleContext) {
+		Map.Entry<String, Map> queryAndParams = buildProcedureQueryWithParams( storedProcedureName, queryParameters );
+		Transaction transaction = transaction( tupleContext );
+		if ( transaction == null ) {
+			DatastoreProvider datastoreProvider = getServiceRegistry().getService( DatastoreProvider.class );
+			BoltNeo4jDatastoreProvider neo4jProvider = (BoltNeo4jDatastoreProvider) datastoreProvider;
+			BoltNeo4jClient client = neo4jProvider.getClient();
+			transaction = client.getDriver().session().beginTransaction();
```

Comment:
Actually I don't know very well the neo4j transaction interface. Can we sure that the transaction will be closed in all cases?

Reply:
Shouldn't the transaction be opened by the client?

Reply:
Thank you for comment, I'll investigate the question for bolt dialect more carefully